workshops/test/test_util.py

```diff
@@ -0,0 +1,138 @@
+# this probably needs refactored for py3
+from datetime import datetime
+from StringIO import StringIO
+from json import dumps
+
+from django.test import TestCase
+
+from ..models import Site, Event, Role, Person
+from ..util import PERSON_UPLOAD_FIELDS, upload_person_task_csv,\
+    verify_upload_person_task
+
+from .base import TestBase
+
+class UploadPersonTaskCSVTestCase(TestCase):
+
+    def compute_from_string(self, csv_str):
+        ''' wrap up buffering the raw string & parsing '''
+        csv_buf = StringIO(csv_str)
+        # compute & return
+        return upload_person_task_csv(csv_buf)
+
+    def test_basic_parsing(self):
+        ''' See views.PERSON_UPLOAD_FIELDS for field ordering '''
+        csv = """personal,middle,family,email\njohn,a,doe,johndoe@email.com\n,jane,a,doe,janedoe@email.com"""
+        person_tasks = self.compute_from_string(csv)
+
+        # assert
+        self.assertEqual(len(person_tasks), 2)
+
+        person = person_tasks[0]
+        for key in ('person', 'role', 'event', 'errors'):
+            self.assertIn(key, person)
+
+        person_dict = person['person']
+        for key in PERSON_UPLOAD_FIELDS:
+            self.assertIn(key, person_dict)
```

Comment:
These two for-loops can be squizzed into:
```python
    self.assertSetEqual(set(('person', 'role', 'event', 'errors')), set(person.keys()))
    self.assertSetEqual(set(PERSON_UPLOAD_FIELDS), set(person_dict.keys()))
```
The advantage is that if `person` or `person_dict` have more keys than we anticipated, we'll get an error.  This ensures that `PERSON_UPLOAD_FIELDS` is up-to-date with `Person` model.